tests/catch/unit/memory/hipHostMalloc.cc

```diff
@@ -185,6 +182,7 @@ TEST_CASE("Unit_hipHostMalloc_NonCoherent") {
                    SYNC_STREAM, ptrType);
   CheckHostPointer(numElements, A, hipEventReleaseToSystem,
                    SYNC_EVENT, ptrType);
+  TestContext::get().cleanContext();
```

Comment:
I think it may be a little tricky to remember to call this clean up function for every test that uses a kernel. Could we tie the state to an object local to each test so that the destructor is always called? or what do you think?